frontend/src/app/api/api/compliance.service.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * H2MS: Hand Hygiene Monitoring System
+ * This is the draft API design for H2MS.
+ *
+ * OpenAPI spec version: 1.0.0
+ *
+ *
+ * NOTE: This class is auto generated by the swagger code generator program.
+ * https://github.com/swagger-api/swagger-codegen.git
+ * Do not edit the class manually.
+ */
+
+/* tslint:disable:no-unused-variable member-ordering */
+
+import {Inject, Injectable, Optional} from '@angular/core';
+import {HttpClient, HttpHeaders, HttpParams} from '@angular/common/http';
+
+import {Observable} from 'rxjs/Observable';
+import '../rxjs-operators';
+
+import {ComplianceReport} from '../model/complianceReport';
+
+import {BASE_PATH, COLLECTION_FORMATS} from '../variables';
+import {Configuration} from '../configuration';
+import {CustomHttpUrlEncodingCodec} from '../encoder';
+
+
+@Injectable()
+export class ComplianceService {
+
+    public defaultHeaders = new HttpHeaders();
+    public configuration = new Configuration();
+    protected basePath = 'https://localhost:8081/api';
+
+    constructor(protected httpClient: HttpClient, @Optional() @Inject(BASE_PATH) basePath: string,
+                @Optional() configuration: Configuration) {
+        if (basePath) {
+            this.basePath = basePath;
+        }
+        if (configuration) {
+            this.configuration = configuration;
+            this.basePath = basePath || configuration.basePath || this.basePath;
+        }
```

Comment:
This gets duplicated later.  If you think this might change then you may want to reduce the duplication.